src/parser/tiger.grm

```diff
@@ -40,37 +48,72 @@
 program :  exp   ()
 
 
-exp:    NIL ()
+exp:  lvalue ()
+    | NIL ()
     | INT ()
+    | STRING ()
     | seqexp ()
     | negation ()
     | callexp ()
-    | infixexp ()
+    | arraycreateexp ()
+    | recordcreateexp ()
+    | assignmentexp ()
+
+    | exp PLUS exp ()
+    | exp MINUS exp ()
+    | exp TIMES exp ()
+    | exp DIVIDE exp ()
+    | exp AND exp ()
+    | exp OR exp ()
+    | exp EQ exp ()
+    | exp NEQ exp ()
+    | exp LT exp ()
+    | exp LE exp ()
+    | exp GT exp ()
+    | exp GE exp ()
+
+    | ifthenelseexp ()
+    | ifthenexp ()
+    | whileexp ()
+    | forexp ()
+    | BREAK ()
+
+negation: MINUS exp ()
 
 seqexp: LPAREN seqexplist RPAREN ()
 
 seqexplist: seqexplist SEMICOLON exp ()
     | exp ()
 
-negation: MINUS exp ()
-
 callexp: ID LPAREN callexplist RPAREN ()
        | ID LPAREN RPAREN ()
 
 callexplist: callexplist COMMA exp ()
     | exp ()
 
-infixexp: exp op exp ()
-
-op: PLUS ()
-    | MINUS ()
-    | TIMES ()
-    | DIVIDE ()
-    | AND ()
-    | OR ()
-    | EQ ()
-    | NEQ ()
-    | LT ()
-    | LE ()
-    | GT ()
-    | GE ()
+arraycreateexp: ID LBRACK exp RBRACK OF exp ()
+
+recordcreateexp: ID LBRACE recordcreatelistexp RBRACE ()
+
+recordcreatelistexp: recordcreatelist ()
+    | ()
+
+fieldcreate: ID EQ exp ()
+
+recordcreatelist: recordcreatelist COMMA fieldcreate ()
+    | fieldcreate ()
+
+assignmentexp: lvalue ASSIGN exp ()
+
+lvalue: ID ()
+    | lvalue DOT ID ()
+    | lvalue LBRACK exp RBRACK ()
+    | ID LBRACK exp RBRACK ()
+
+ifthenelseexp: IF exp THEN exp ELSE exp ()
+
+ifthenexp: IF exp THEN exp ()
```

Comment:
I'm pretty sure this just makes it a syntax error. From what I'm reading anyway.